tensorboard/summary/tf_summary_test.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Integration test for exporting TB symbols under the `tf.summary` API.
+
+This checks the integration between TF and TB to verify that the mechanism we
+use to export TensorBoard symbols as part of the `tf.summary` API works.
+
+We also test this at an even higher level (and with many different import
+sequences) via our pip package build script. But that script is run only in
+our external CI, whereas this python test can run anywhere our normal tests run.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import sys
+import unittest
+
+class TfSummaryExportTest(unittest.TestCase):
+
+  def test_tf_summary_export(self):
+    # Check as a precondition that TF wasn't already imported.
+    self.assertEqual('notfound', sys.modules.get('tensorflow', 'notfound'))
```

Comment:
Why is this needed, given that we immediately import TensorFlow?

The existing occurrence of this assertion in `summary_test.py` ensures
that TensorFlow isn’t automatically imported by accessing certain
symbols, but I don’t see why we need something like that here.

Reply:
I think this is a stronger test if we ensure that it's testing the entire flow of importing TF and then using tf.summary, versus having the import happen "somewhere else".  This check enforces that we don't modify this test down the road to import TF at the top of the file or in another test method, which would muddy state going into this test.

I've elaborated the comment to explain this reasoning better.

Reply:
Okay, fine with me. Thanks.